minetest/minetest
# Make shading of CAOs optional

Second fix for #9985. The goal is to make shading optional for CAOs. To do this we introduce boolean property `shaded` which is `true` by default.

## To do

Ready for Review

- [X] Bump protocol version (supposedly not required anymore)
- [x] Update API doc

## How to test

In games/devtest/mods/testentities/visuals.lua add following entities

```lua
minetest.register_entity("testentities:mesh2", {
	initial_properties = {
		visual = "mesh",
		mesh = "testnodes_pyramid.obj",
		tiles = {"testnodes_mesh_stripes.png"},
		shaded = true,
	},
})

minetest.register_entity("testentities:mesh2_unshaded", {
	initial_properties = {
		visual = "mesh",
		mesh = "testnodes_pyramid.obj",
		tiles = {"testnodes_mesh_stripes.png"},
		shaded = false,
	},
})
```

Launch the game and add them using Entity Spawner. One should be shaded and the other one should be flat.
